src/bst.py

```diff
@@ -0,0 +1,207 @@
+                    return False
+                else:
+                    parent = parent.right
+                    return self._compare_nodes(parent)
+            elif self.value < parent.value:
+                if parent.left is None:
+                    return False
+                else:
+                    parent = parent.left
+                    return self._compare_nodes(parent)
+        else:
+            return True
+
+    def get_dot(self):
+        """return the tree with root 'self' as a dot graph for visualization"""
+        return "digraph G{\n%s}" % ("" if self.value is None else (
+            "\t%s;\n%s\n" % (
+                self.value,
+                "\n".join(self._get_dot())
+            )
+        ))
+
+    def _get_dot(self):
+        """recursively prepare a dot graph entry for this node."""
+        if self.left is not None:
+            yield "\t%s -> %s;" % (self.value, self.left.value)
+            for i in self.left._get_dot():
+                yield i
+        elif self.right is not None:
+            r = random.randint(0, 1e9)
+            yield "\tnull%s [shape=point];" % r
+            yield "\t%s -> null%s;" % (self.value, r)
+        if self.right is not None:
+            yield "\t%s -> %s;" % (self.value, self.right.value)
+            for i in self.right._get_dot():
+                yield i
+        elif self.left is not None:
+            r = random.randint(0, 1e9)
+            yield "\tnull%s [shape=point];" % r
+            yield "\t%s -> null%s;" % (self.value, r)
+
+
+class Bst(object):
+    """Define binary search tree class."""
+
+    def __init__(self, iterable=None):
+        """
+        Create an instance of binary search tree.
+        """
+        self.head = None
+        self.counter = 0
+        if iterable is not None:
+            try:
+                for item in iterable:
+                    self.insert(item)
+            except TypeError:
+                self.insert(iterable)
+
+    def insert(self, value):
+        """
+        Insert the value into bst. If value is already present,
+        it will be ignored.
+        """
+        new_node = Node(value)
+        if self.head is None:
+            self.head = new_node
+        else:
+            self.head._insert_node(new_node)
+        self.counter += 1
+
+    def contains(self, value):
+        """
+        Returns True if the value in the bst, False if not.
+        """
+        new_node = Node(value)
+        if self.head is None:
+            return False
+        else:
+            if new_node._compare_nodes(self.head):
+                return True
+            else:
+                return False
+
+    def size(self):
+        """
+        Return the integer size of the bst.
+        Return zero if the bst is empty.
+        """
+        return self.counter
+
+    def depth(self):
+        """
+        Return an integer representing the total levels in a tree.
+        """
+        return self.head.depth
```

Comment:
What's going to happen here if the Tree is totally empty? Make sure you test this case for all of your unit tests.